addons/refuel/functions/fnc_handleRespawn.sqf

```diff
@@ -0,0 +1,27 @@
+/*
+ * Author: Dystopian
+ * Clean variables on unit respawn.
+ *
+ * Arguments:
+ * 0: Unit <OBJECT>
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ * [player] call ace_refuel_fnc_handleRespawn
+ *
+ * Public: No
+ */
+#include "script_component.hpp"
+
+params ["_unit"];
+TRACE_1("respawn",_unit);
+
+if (isNull (_unit getVariable [QGVAR(nozzle), objNull])) exitWith {};
+
+[_unit, "forceWalk", "ACE_refuel", false] call EFUNC(common,statusEffect_set);
```

Comment:
statusEffect should take care of respawns automatically
or was this explicitly needed?

Reply:
it was in the original `XEH_respawn.sqf`, I didn't check. I will check now and delete if yes